kong/init.lua

```diff
@@ -337,7 +377,9 @@ function Kong.init()
     certificate.init()
   end
 
-  mesh.init()
+  if kong.configuration.database ~= "off" then
```

Comment:
this is not disallowing service mesh. For DB-less, `mesh.init()` is just called from a different place because of the different order things need to happen.

Reply:
What is the different order required? It was placed here because some plugins needed mesh initialised before they could load.

Reply:
We can only have entities in DB-less after we have a cache, and we only have the cache in `init_worker`. When mesh is used, the `cluster_ca` object needs to be fed in the `config.yml` file (as the nodes don't talk to each other, there's no DB or cluster_events).

Reply:
The "cluster_ca" entity is singular and global, does that limitation still apply?

Reply:
yes, because the cache is the backing store for db-less. Which plugin you have in mind that won't work with the current implementation?